src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -240,6 +239,50 @@ public override CSharpType FindTypeForSchema(Schema schema)
             _ => throw new NotImplementedException()
         };
 
+        /// <summary>
+        /// Looks for an <see cref="OperationGroup" /> by a <see cref="MgmtRestClient" />.
+        /// </summary>
+        /// <param name="restClient">The <see cref="MgmtRestClient" /> instance which is built from the operation group.</param>
+        /// <returns>The operation group. `null` if not found.</returns>
+        public OperationGroup FindOperationGroup(MgmtRestClient restClient) {
+            return EnsureRestClients().FirstOrDefault(ogToClient => ogToClient.Value == restClient).Key;
+        }
+
+        /// <summary>
+        /// Looks for an <see cref="OperationGroup" /> by a <see cref="ResourceOperation" />.
+        /// </summary>
+        /// <param name="resourceOperation">The <see cref="ResourceOperation" /> instance which is built from the operation group.</param>
+        /// <returns>The operation group. `null` if not found.</returns>
+        public OperationGroup FindOperationGroup(ResourceOperation resourceOperation) {
+            return EnsureResourceOperations().FirstOrDefault(ogToOperation => ogToOperation.Value == resourceOperation).Key;
+        }
+
+        /// <summary>
+        /// Looks for an <see cref="OperationGroup" /> by <see cref="ResourceContainer" />.
+        /// </summary>
+        /// <param name="container">The <see cref="ResourceContainer" /> instance which is built from the operation group.</param>
+        /// <returns>The operation group. `null` if not found.</returns>
+        public OperationGroup FindOperationGroup(ResourceContainer container) {
+            return EnsureResourceContainers().FirstOrDefault(ogToContainer => ogToContainer.Value == container).Key;
+        }
+
+        /// <summary>
+        /// Looks for an <see cref="OperationGroup" /> by <see cref="ResourceData" />.
+        /// </summary>
+        /// <param name="resourceData">The <see cref="ResourceData" /> instance which is built from the operation group.</param>
+        /// <returns>The operation group. `null` if not found.</returns>
+        public OperationGroup FindOperationGroup(ResourceData resourceData) {
+            return EnsureResourceData().FirstOrDefault(ogToData => ogToData.Value == resourceData).Key;
```

Comment:
You can make the OperationGroup property public: https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs#L30 and avoid this roundtrip.

Reply:
Agreed, doesn't look like these are even used so I will remove them for this PR